src/object-diagram/utilities/export/fileSaver.ts

```diff
@@ -0,0 +1,26 @@
+import { writeFile } from 'fs/promises';
+import { homedir } from 'os';
+import { dirname } from 'path';
+import { window } from 'vscode';
+import { MementoAccessor } from '../storage/mementoAccessor';
+import { SaveDialogOptionsFactory } from './saveDialogOptionsFactory';
+
+export class FileSaver {
+  constructor(
+    private readonly lastDirectoryAccessor: MementoAccessor<string>,
+    private readonly saveDialogOptionsFactory: SaveDialogOptionsFactory,
+    private readonly fileNameStub: string,
+    private readonly dataFactory: () => string
+  ) {}
+
+  async saveFile(): Promise<void> {
+    const uri = await window.showSaveDialog(
+      this.saveDialogOptionsFactory.createSaveDialogOptions(this.lastDirectoryAccessor.getValue() ?? homedir(), this.fileNameStub)
+    );
+    if (uri) {
+      const { fsPath } = uri;
```

Comment:
```suggestion
      cosnt fsPath = url.fsPath;
```
Don't you think that's easier to read?

Reply:
I'm very used to read object destructuring assignments. It's even my preferred way to write assignments, especially when there are no naming conflicts and when the identifiers are short.